File: src/pt/joaomneto/titancompanion/adventure/impl/STRIDERAdventure.java
package pt.joaomneto.titancompanion.adventure.impl;

import android.os.Bundle;
import android.view.Menu;

import java.io.BufferedWriter;
import java.io.IOException;

import pt.joaomneto.titancompanion.R;
import pt.joaomneto.titancompanion.adventure.Adventure;

public class STRIDERAdventure extends Adventure {

	Integer currentFear;
	Integer time;
	Integer oxygen;

	private static int fragmentIdx = 0;

	private final static int FRAGMENT_VITAL_STATS = fragmentIdx++;
	private final static int FRAGMENT_VITAL_TIME_OXYGEN = fragmentIdx++;
	private final static int FRAGMENT_COMBAT = fragmentIdx++;
	private final static int FRAGMENT_EQUIPMENT = fragmentIdx++;
	private final static int FRAGMENT_NOTES = fragmentIdx++;


	public STRIDERAdventure() {
		super();
		fragmentConfiguration.clear();
		fragmentConfiguration.put(FRAGMENT_VITAL_STATS, new AdventureFragmentRunner(R.string.vitalStats,
				"pt.joaomneto.titancompanion.adventure.impl.fragments.strider.STRIDERAdventureVitalStatsFragment"));
		fragmentConfiguration.put(FRAGMENT_VITAL_TIME_OXYGEN, new AdventureFragmentRunner(R.string.vitalStats,
				"pt.joaomneto.titancompanion.adventure.impl.fragments.strider.STRIDERAdventureTimeOxygenFragment"));
		fragmentConfiguration.put(FRAGMENT_COMBAT, new AdventureFragmentRunner(R.string.fights,
				"pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_EQUIPMENT, new AdventureFragmentRunner(R.string.goldEquipment,
				"pt.joaomneto.titancompanion.adventure.impl.fragments.strider.STRIDERAdventureEquipmentFragment"));
		fragmentConfiguration.put(FRAGMENT_NOTES, new AdventureFragmentRunner(R.string.notes,
				"pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment"));
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		try {
			super.onCreate(savedInstanceState);

		} catch (Exception e) {
			e.printStackTrace();
		}

	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		getMenuInflater().inflate(R.menu.adventure, menu);
		return true;
	}

	@Override
	public void storeAdventureSpecificValuesInFile(BufferedWriter bw) throws IOException {

		bw.write("fear=" + currentFear + "\n");
		bw.write("time=" + time + "\n");
		bw.write("oxygen=" + oxygen + "\n");
	}

	public Integer getCurrentFear() {
		return currentFear;
	}

	public void setCurrentFear(Integer currentFear) {
		this.currentFear = currentFear;
	}


	@Override
	protected void loadAdventureSpecificValuesFromFile() {
		setCurrentFear(Integer.valueOf(getSavedGame().getProperty("fear")));
		setTime(Integer.valueOf(getSavedGame().getProperty("time")));
		setOxygen(Integer.valueOf(getSavedGame().getProperty("oxygen")));
	}

	public Integer getTime() {
		return time;
	}

	public void setTime(Integer time) {
		this.time = time;
	}

	public void increaseTime() {
		this.time = Math.min(time+1, 48);
	}

	public void decreaseTime() {
		this.time = Math.max(time-1, 0);
	}

	public Integer getOxygen() {
		return oxygen;
	}

	public void setOxygen(Integer oxygen) {
		this.oxygen = oxygen;
	}

	public void increaseOxygen() {
		this.oxygen = Math.min(oxygen+1, 20);
	}

	public void decreaseOxygen() {
		this.oxygen = Math.max(oxygen-1, 0);
	}


}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/STRIDERAdventure.kt
package pt.joaomneto.titancompanion.adventure.impl

import android.view.Menu
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventure.Adventure
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureCombatFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.strider.STRIDERAdventureEquipmentFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.strider.STRIDERAdventureTimeOxygenFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.strider.STRIDERAdventureVitalStatsFragment
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner
import pt.joaomneto.titancompanion.util.DiceRoller
import java.io.BufferedWriter
import java.io.IOException

class STRIDERAdventure : Adventure(
    arrayOf(
        AdventureFragmentRunner(
            R.string.vitalStats,
            STRIDERAdventureVitalStatsFragment::class
        ),
        AdventureFragmentRunner(
            R.string.timeAndOxygen,
            STRIDERAdventureTimeOxygenFragment::class
        ),
        AdventureFragmentRunner(
            R.string.fights,
            AdventureCombatFragment::class
        ),
        AdventureFragmentRunner(
            R.string.equipment2,
            STRIDERAdventureEquipmentFragment::class
        ),
        AdventureFragmentRunner(
            R.string.notes,
            AdventureNotesFragment::class
        )
    )
) {

    var currentFear: Int = -1
    var time: Int = -1
    var oxygen: Int = -1

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.adventure, menu)
        return true
    }

    @Throws(IOException::class)
    override fun storeAdventureSpecificValuesInFile(bw: BufferedWriter) {

        bw.write("fear=" + currentFear + "\n")
        bw.write("time=" + time + "\n")
        bw.write("oxygen=" + oxygen + "\n")
    }

    override fun loadAdventureSpecificValuesFromFile() {
        currentFear = Integer.valueOf(savedGame.getProperty("fear"))
        time = Integer.valueOf(savedGame.getProperty("time"))
        oxygen = Integer.valueOf(savedGame.getProperty("oxygen"))
    }

    fun testFear() {

        val result = DiceRoller.roll2D6().sum <= currentFear

        val message = if (result) R.string.success else R.string.failed
        Adventure.Companion.showAlert(message, this)

        (getFragment(STRIDERAdventureVitalStatsFragment::class))?.refreshScreensFromResume()
    }

    fun increaseTime() {
        this.time = Math.min(time + 1, 48)
    }

    fun decreaseTime() {
        this.time = Math.max(time - 1, 0)
    }

    fun increaseOxygen() {
        this.oxygen = Math.min(oxygen + 1, 20)
    }

    fun decreaseOxygen() {
        this.oxygen = Math.max(oxygen - 1, 0)
    }

    companion object {

        private val FRAGMENT_VITAL_TIME_OXYGEN = 1
        private val FRAGMENT_COMBAT = 2
        private val FRAGMENT_EQUIPMENT = 3
        private val FRAGMENT_NOTES = 4
    }
}
